### PR TITLE
Implement parsing the index file and the `ls-files` command

### PR DESCRIPTION
**Pull request by mkpro118**
## Summary by CodeRabbit

- **New Features**
  - Introduced the `ls-files` command, allowing users to list files tracked in the repository index with options for full paths, debug metadata, and custom output separators.

- **Bug Fixes**
  - Improved handling of empty or missing index files to ensure graceful behavior in the `ls-files` command.

- **Documentation**
  - Updated the README to mark the `ls-files` command as complete and adjusted text formatting.

- **Tests**
  - Added comprehensive tests for the `ls-files` command, covering various output options and edge cases.

### CODE REVIEW

**Comment by mkpro118** [comment]
@coderabbitai my tests defined in this branch are failing on windows. Any idea why? please suggest improvements or point out mistakes where I am not accounting to windows specific behavior. please read "tests/core/test_ls_files.rs" and let me know what may be the issue

File: tests/core/test_ls_files.rs [hunk 1]
```diff
@@ -0,0 +1,519 @@
...
+            let args: [&[&str]; 1] = [&[]];
+            let namespace = make_namespaces(&args).next().unwrap();
+
+            ls_files(&namespace)
+        });
+
+        assert!(result.is_ok(), "ls-files should succeed with basic index");
+
+        let output = result.unwrap();
+        println!("test_ls_files_basic_functionality\n\n--------\n{output:?}--------\n");
```
Comment (mkpro118) [inline tests/core/test_ls_files.rs:244]: @coderabbitai This is returning
```
test_ls_files_basic_functionality
--------
""--------
```

and similar output for every other test. like the ls_files command just returns no output.
please read src/core/commands/ls_files.rs and see what the issue might be.

Comment (mkpro118) [reply tests/core/test_ls_files.rs:244]: @coderabbitai these tests literally works perfectly on my local windows machine. so I do not know why exactly it fails in github actions

**Comment by mkpro118** [comment]
AI was completely useless debugging this 😐
